builtin/credential/ldap/backend.go

```diff
@@ -227,213 +194,6 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 	return policies, ldapResponse, allGroups, nil
 }
 
-/*
- * Parses a distinguished name and returns the CN portion.
- * Given a non-conforming string (such as an already-extracted CN),
- * it will be returned as-is.
- */
-func (b *backend) getCN(dn string) string {
-	parsedDN, err := ldap.ParseDN(dn)
-	if err != nil || len(parsedDN.RDNs) == 0 {
-		// It was already a CN, return as-is
-		return dn
-	}
-
-	for _, rdn := range parsedDN.RDNs {
-		for _, rdnAttr := range rdn.Attributes {
-			if rdnAttr.Type == "CN" {
-				return rdnAttr.Value
-			}
-		}
-	}
-
-	// Default, return self
-	return dn
-}
-
-/*
- * Discover and return the bind string for the user attempting to authenticate.
- * This is handled in one of several ways:
- *
- * 1. If DiscoverDN is set, the user object will be searched for using userdn (base search path)
- *    and userattr (the attribute that maps to the provided username).
- *    The bind will either be anonymous or use binddn and bindpassword if they were provided.
- * 2. If upndomain is set, the user dn is constructed as 'username@upndomain'. See https://msdn.microsoft.com/en-us/library/cc223499.aspx
- *
- */
-func (b *backend) getUserBindDN(cfg *ConfigEntry, c *ldap.Conn, username string) (string, error) {
```

Comment:
This was moved without any change. It was added to the client because it makes external calls. It also had to be exported so it could be used outside the package.